### PR TITLE
avm1: Fix some cases of audio looping or not stopping

### LINKED ISSUE #963: Samorost: Strange behaviours after switching movies
In Samorost, each level exists as a different movie file. When switching to the next level, the background and music are meant to change. However, in Ruffle, the previous background and music track overlap with the next one.
Each level works properly if loaded independently.

**Video**: https://streamable.com/l4b2jz
**Video (rybar.swf, by itself)**: https://streamable.com/u3yd8p
**Game file**: https://amanita-design.net/samorost-1/

### LINKED ISSUE #2473: [NSFW] Problems switching on/off sound or music in a flash, sounds can get badly layered
http://eye.swfchan.com/flash.asp?id=239119&n=Uzaki-Chan+Not+Here+Not+Now%21.swf

1) Listener is blasted with a sound if he tries to meddle with Music on/off, Ambience on/off switches, probably because the sounds are played at the max volume (sound volume is unimplemented yet AFAIK)
image

2) If you repeatedly (more than once) press Music off or Ambience off button, the sounds will layer depending on how many times you pressed the button, and it basically becomes an earrape (and its stoppable only if you turn down the volume/switch the sound on the tab)

### LINKED ISSUE #7447: Sound loops continue to play https://hoogerbrugge.com/ml/
[Modern Living](https://hoogerbrugge.com/ml/) consists of 100 interactive animations.

Navigated via a vertical numbered nav bar populated by thumbnails of each animation.

Many consist of a background looping sound, which plays infinitely while you are viewing each animation in the main view port.

However, the background looping audio should stop when another animation is clicked on.

This becomes more obvious the more animations you click on and each background looping audio layers on top of each other.

### Expected behavior

Expected behaviour is that all sounds from a previous animation should stop if you click on another animation from the main navigation

### Affected platform

Browser's extension

### Operating system

MacOS Monterey 12.3

### Browser

Brave 1.41.96 Chromium 103.0.5060.114 (arm64)

### Additional information

Another test URL on the same domain with this issue is: [https://hoogerbrugge.com/nails/](https://hoogerbrugge.com/nails/) 

Thank you for this project!

### LINKED ISSUE #9795: avm1: sound does not stop
### Describe the bug

the bug occurs when you start the game (i believe its #[2914](https://github.com/ruffle-rs/ruffle/issues/2914)) and there's another bug when pressing space in game 

[game.zip](https://github.com/ruffle-rs/ruffle/files/10860661/game.zip)

(Video)
https://user-images.githubusercontent.com/56921534/222135251-11cabb66-7650-47c7-a07f-f4cb38536050.mp4

### Expected behavior

audio should stop after the game loads and the sound should also stop after releasing space and the arm retracts

(Video)
https://user-images.githubusercontent.com/56921534/222135351-995a8f74-6616-445e-8aa8-c4000b475ca4.mp4

### Affected platform

Desktop app

### Operating system

Windows 11

### Browser

_No response_

### Additional information

to run the game simply run bootstrap.swf

### LINKED ISSUE #14663: webagent007: Music does not stop
### Describe the bug

On the [old webagent007.com Flash site](http://www.wa007.com/2000/newsite/main.html), when a new music track plays, the previous track does not stop. So multiple audio tracks end up layering over each other.

### Expected behavior

When one music track starts, the previous one should stop.

### Content Location

Page URL: http://www.wa007.com/2000/newsite/main.html
SWF URL: http://www.wa007.com/2000/newsite/interface1.swf
Note: The site uses multiple assets, so the SWF needs to be loaded from the provided URL (not downloaded).

### Affected platform

Browser's extension

### Operating system

Windows 10

### Browser

Firefox 121

### Additional information

Probably related to #9795.

### LINKED ISSUE #16947: Audio not stopping / continues tracks.. AS1, AS2, AS3
### Describe the bug

For some reason audio tends to never stop in certain areas of the game.. its almost as if the track isnt being told to stop. This not only happens on AS1-AS2 but AS3 as well.. 

### Expected behavior

Audio to stop when exiting games and or a room.

### Content Location

AS1-AS2 SWF Versions:
https://git.solero.me/solero/legacy-media
AS2-AS3 SWF Versions:
https://git.solero.me/solero/vanilla-media
Game Client:
https://github.com/solero/houdini
AS1-AS2 Demo (Legacy Media) :
https://play.cporiginal.net
AS3 Demo (Vanilla Media) :
https://beta.cporiginal.net
^^^^^^^^^^^^^^^^^
For testing you can use these login details:
Username: TestAccount
Password: TestAccount32

### Affected platform

Browser's extension

### Operating system

Windows 11

### Browser

Google Chrome Version 126.0.6478.127 (Official Build) (64-bit)

### Additional information

The client runs off of TCP connection, so to solve that issue there is a Websockify Proxy setup to connect Websockets to the TCP connection that has to be established to communicate to the game client.

### PR DESCRIPTION
**Pull request by Dinnerbone**
Definitely not every case, but I think we have a lot of issues to work through here.

Slightly impossible to add tests for this :(

Fixes #9795
Fixes #7447
Fixes #14663
Fixes #963
Fixes #2473
Fixes #16947

### CODE REVIEW

**Comment by danielhjacobs** [comment]
Have you tested https://github.com/ruffle-rs/ruffle/issues/963?

**Comment by Dinnerbone** [comment]
> Have you tested #963?

Seems to be fixed too

**Comment by danielhjacobs** [comment]
> Seems to be fixed too

Then this likely supersedes https://github.com/ruffle-rs/ruffle/pull/13781

File: core/src/backend/audio.rs [hunk 1]
```diff
@@ -32,6 +32,7 @@ mod decoders {
     }
 }
 
+use ::swf::{CharacterId, SoundInfo};
```
Comment (sleepycatcoding) [inline core/src/backend/audio.rs:35]: Why does this import start with `::`?

Comment (Dinnerbone) [reply core/src/backend/audio.rs:35]: I have no idea... huh.

I didn't know this was valid syntax? :D

Comment (Dinnerbone) [reply core/src/backend/audio.rs]: Fixed!

**Review by adrian17 — APPROVED** [review]

**Pull request merged**